docs/adr/0001-user-roles.rst

```diff
@@ -0,0 +1,118 @@
+ADR 0001 - User roles
+=====================
+
+Status
+------
+
+Proposed
+
+
+Context
+-------
+
+Depending on what we expect them to do on Marsha, we need to define some user roles.
+
+First we have the people managing a Marsha instance.
+
+Then we have people linking their own website (this website is a "consumer site") to a Marsha instance, to host videos.
+
+These consumer sites can host many customers. We call these organizations. And these organizations have managers that can administrate some things about authors, video sharing between authors...
+
+Next we have the video authors, belonging the organizations.
+
+And finally we have the users coming to watch videos.
+
+
+Decision
+--------
+
+Let's separate those 5 roles:
+
+"staff"
++++++++
+
+Purpose
+~~~~~~~
+
+To manage a Marsha instance
+
+Implementation
+~~~~~~~~~~~~~~
+
+These are simply instances of the Django ``User`` model, with the flag ``is_staff`` set to ``True``.
+
+"admins"
+++++++++
+
+Purpose
+~~~~~~~
+
+To manage the link between a consumer site and a Marsha instance, and the organizations of this consumer site on the instance
+
+Implementation
+~~~~~~~~~~~~~~
+
+To represent a consumer site on an Marsha instance, we have a ``ConsumerSite`` Django model. With a ``ManyToMany`` link to the ``User`` model, named ``admins`` (not a single admin, to avoid having no admin if the only one existing is not available anymore)).
+
+
+"managers"
+++++++++++
+
+Purpose
+~~~~~~~
+
+To manage the authors in the organization. And allow videos to be private to authors or public for all authors.
+
+Implementation
+~~~~~~~~~~~~~~
+
+To represent an organization on a Marsha instance, we have a ``Organization`` Django model. With a ``ManyToMany` link to the ``User`` model, named ``managers``.
+
+
+"authors"
++++++++++
+
+Purpose
+~~~~~~~
+
+To post videos on a Marsha instance to be used on the consumer site.
+
+Implementation
+~~~~~~~~~~~~~~
+
+An author is simply an instance of the ``User`` model, but has a link to an ``Organization`` via a ``ManyToMany`` link, named ``organizations``.
+
+They cannot manage anything directly on a ``Marsha`` instance. Every action they can do are done via a link/api/iframe/wathever from the consumer site, with a token identifying them. This token must include a way to identify the author (for example its ID on the consumer site) and the organization for which he wants to do the current action (upload/update/pick a video...).
```

Comment:
as discussed, as a first approach we can require them to do everything on Marsha.

Reply:
Yes I am currently rewriting some parts following our phone discussion